posydon/binary_evol/CE/step_CEE.py

```diff
@@ -814,4 +826,92 @@ def CEE_simple_alpha_prescription(
+                    print("which is the max of RLO1 or RLO2 of the inner cores: ", separation_for_inner_RLO1 ,  separation_for_inner_RLO2, "in Rsun")
+
+                if not double_CE:
+                    # Ebind(Mejected) = a_CE*D_orb =>
+                    # GMMej/(lambda * R)  = a_CE * [ - GM1M2/(2a_i) +  G(M1-Mej)*M2/(2a_f) ]
+                    # with a_f = separation_postCEE
+                    # and solving it for Mejected
+                    Mejected_donor = alpha_CE * (m1_i * const.Msun * m2_i * const.Msun)/2. * (1./separation_postCEE - 1./separation_i) \
+                              / ( (m1_i * const.Msun/(lambda1_CE*radius1 * const.Rsun)) + ((alpha_CE*m2_i * const.Msun)/(2.*separation_postCEE))  )
+
+                    Mejected_donor = Mejected_donor / const.Msun # to go from cgs to solar masses again
+                    Mejected_comp = 0.0
+
+                    if Mejected_donor > m1_i  - mc1_i:
+                        Mejected_donor = (m1_i  - mc1_i) -0.01 # at least this value of envelope is left.
+                        warnings.warn("Mejected_donor in normal CEE is found to be more that the initial envelope. Reduced to initial_envelope - 0.01 Msun")
+
+                else: # in double_CE
+
+                    # Assuming that the ratio of ejected mass of each (common) envelope
+                    # is the same as the ratio of the initial envelope masses:
+                    # the root of the mass lost for each is the solution of a quadratic algebraic equation
+
+                    WF = (m1_i  - mc1_i)/ (m2_i - mc2_i) # weight factor:
+                                                        # = Mdonor,envelope / Mcomp,envelope == Mejected_donor / Mejected_comp
+
+                    # Mejected_comp = Mejected_donor / WF
+                    # A*Mejected_donor^2 + B*Mejected_donor + C = 0
+
+                    A = alpha_CE / (WF*2.*separation_postCEE)
+                    B = ( ( - alpha_CE*m1_i * const.Msun/ (WF*2.*separation_postCEE)) - (alpha_CE*m2_i * const.Msun/ (2.*separation_postCEE)) \
+                        - (m1_i * const.Msun/ (lambda1_CE*radius1 * const.Rsun)) - (m2_i * const.Msun/ (WF*lambda2_CE*radius2 * const.Rsun)))
+                    C = alpha_CE * (m1_i * const.Msun * m2_i * const.Msun)/2. * (1./separation_postCEE - 1./separation_i)
+                    roots = np.roots([A,B,C])
+                    if min( np.real(roots[0]), np.real(roots[1]) ) < 0.:
+                        root = max( np.real(roots[0]), np.real(roots[1]) ) # if one root is negative, the other is considered the physical solution
+                    elif max( np.real(roots[0]), np.real(roots[1]) ) > m1_i  - mc1_i: # if one root is more positive than the Menvelope that we allow to ejected
+                                                                                     # then the other is considered the physical solution
+                        root = min( np.real(roots[0]), np.real(roots[1]) )
+
```

Comment:
We need an `else` clause here, otherwise `root` could be undefined.
I'd suggest to use the minimum in there.